src/components/withCurrentUserPersonalDetails.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {ComponentType, RefAttributes, ForwardedRef, useMemo} from 'react';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import ONYXKEYS from '../ONYXKEYS';
+import personalDetailsPropType from '../pages/personalDetailsPropType';
+import type {PersonalDetails, Session} from '../types/onyx';
+
+type CurrentUserPersonalDetails = PersonalDetails | Record<string, never>;
+
+type WithCurrentUserPersonalDetailsProps = {
+    currentUserPersonalDetails: CurrentUserPersonalDetails;
+};
+
+type WithCurrentUserPersonalDetailsComponentProps = {
+    /** Personal details of all the users, including current user */
+    personalDetails: Record<string, PersonalDetails> | null;
+
+    /** Session of the current user */
+    session: OnyxEntry<Session>;
+};
```

Comment:
I would make the names shorter for better visibility:
```suggestion
type OnyxProps = {
    /** Personal details of all the users, including current user */
    personalDetails: OnyxEntry<Record<string, PersonalDetails>>;

    /** Session of the current user */
    session: OnyxEntry<Session>;
};

type ComponentProps = OnyxProps & {
    currentUserPersonalDetails: CurrentUserPersonalDetails;
};
```